services/api/service.go

```diff
@@ -1595,6 +1663,620 @@ func (api *RelayAPI) checkBuilderEntry(w http.ResponseWriter, log *logrus.Entry,
+	}
+	txs, err := common.DecodeTransactions(transactionBytes)
+	if err != nil {
+		log.WithError(err).Warn("could not decode transactions")
+		api.RespondError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	if len(txs) == 1 {
+		log.Error("We require a payment tx along with the TOB txs!")
+		api.Respond(w, http.StatusBadRequest, "We require a payment tx along with the TOB txs!")
+		return
+	}
+	log.Info("DEBUG: decoded txs are ", txs)
+
+	// TODO - check the validity of txs
+
+	err = api.checkTxAndSenderValidity(txs, log)
+	if err != nil {
+		log.WithError(err).Warn("error validating the txs")
+		api.RespondError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	lastTx := txs[len(txs)-1]
+
+	// TODO - simulate the tx to check for tx validity
+	err = api.checkTobTxsStateInterference(txs)
+	if err != nil {
+		log.WithError(err).Warn("could not validate tob txs")
+		api.RespondError(w, http.StatusInternalServerError, err.Error())
+	}
+
+	tx := api.redis.NewTxPipeline()
+
+	tobTxValue := lastTx.Value()
+	log.Info("DEBUG: tobTxValue is ", tobTxValue)
+	log.Info("DEBUG: Writing TobTxValue to redis cache with key", slot, parentHash)
+	// store it in a redis cache. The ROB block will pick it up and assemble it along with its own txs
+	currentTobTxValue, err := api.redis.GetTobTxValue(context.Background(), tx, slot, parentHash)
+	if err != nil {
+		log.WithError(err).Warn("could not get current Tob Tx value")
+		api.RespondError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	log.Info("DEBUG: currentTobTxValue is ", currentTobTxValue)
+	log.Info("DEBUG: tobTxValue is ", tobTxValue)
+	log.Info("DEBUG: tobTxValue.Cmp(currentTobTxValue) is ", tobTxValue.Cmp(currentTobTxValue))
+	log.Info("DEBUG: Wrote TobTxValue to redis cache with key", slot, parentHash)
+
+	if currentTobTxValue.Cmp(tobTxValue) > 0 {
+		log.Error("TOB tx value is less than the current value!")
+		api.Respond(w, http.StatusBadRequest, "TOB tx value is less than the current value!")
+		return
+	}
+
+	// TODO - bchain - simulate the txs on the parent block. If it fails, reject the txs.
+	// This is already solved and should be straightforward to implement. its basically simulation
+	// with some additional checks
+
+	// add the tob tx to the redis cache
+	err = api.redis.SetTobTx(context.Background(), tx, slot, parentHash, transactionBytes)
+	if err != nil {
+		log.WithError(err).Warn("could not set Tob Tx")
+		api.RespondError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	// set the tob tx value in the redis cache
+	err = api.redis.SetTobTxValue(context.Background(), tx, tobTxValue, slot, parentHash)
+	if err != nil {
+		log.WithError(err).Warn("could not set Tob Tx Value")
+		api.RespondError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	api.Respond(w, http.StatusOK, "Tob Tx submitted successfully!")
+	return
+}
+
+func (api *RelayAPI) handleSubmitNewRobBlock(w http.ResponseWriter, req *http.Request) {
```

Comment:
any chance we can use handleSubmitNewBlock out of the box. hard to review when i can't see the diff between this function and the original